appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,22 +155,38 @@ export class CertificateManager {
         }
     }
 
-    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig): Promise<WebhookCertData> {
+    public async GetSecretDetails(operationId: string, kubeConfig: k8s.KubeConfig, clusterArmId: string, clusterArmRegion: string): Promise<WebhookCertData> {
         try {
             const k8sApi = kubeConfig.makeApiClient(k8s.CoreV1Api);
             const secretsObj: k8s.V1Secret = await k8sApi.readNamespacedSecret({ name: CertificateStoreName, namespace: KubeSystemNamespaceName });
+            let certificate: WebhookCertData = null;
             if (secretsObj.data) {
-                const certificate: WebhookCertData = {
-                    caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
-                    caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
-                    tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
-                    tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
-                };
+                try {
```

Comment:
This is very confusing, why is this try nested into the first try? Let's have two separate tries in sequence - one for the API call, one for interpreting its results. As-is is very confusing.